### PR TITLE
Check project directory is correct and ensure there's a go.mod file

### PR DESCRIPTION
**Pull request by rafaelDev0ps**
this pull request is related to the issue #2 
1. check if the project directory matches with project name
```go
func matchProjectDirectory() bool {
	wd := process.GetWorkingDir()
	dirs := strings.Split(wd, "/")

	projectDir := dirs[len(dirs)-1]

	if GetModuleName() == projectDir {
		return true
	}
	return false
}
```
2. make sure go.mod file of the project created exists
```go
func HasModuleFile(path string) bool {
	_, err := os.Stat(path)
	return !os.IsNotExist(err)
}
```

### CODE REVIEW

**Comment by maddalax** [comment]
great, I'll try to look at this later today or early tomorrow, thanks!

**Comment by rafaelDev0ps** [comment]
np @maddalax, let me know if there's something wrong or something that could be enhanced!

File: cli/htmgo/tasks/astgen/entry.go [hunk 1]
```diff
@@ -402,6 +414,18 @@ func GetModuleName() string {
 	return modName
 }
 
+func matchProjectDirectory() bool {
+	wd := process.GetWorkingDir()
+	dirs := strings.Split(wd, "/")
+
+	projectDir := dirs[len(dirs)-1]
+
+	if GetModuleName() == projectDir {
```
Comment (maddalax) [inline cli/htmgo/tasks/astgen/entry.go:423]: I actually don't believe its required these match, for example: the framework module in htmgo is 
```
github.com/maddalax/htmgo/framework
```

but the project dir path would be just `framework`

which would make this fail. Can we remove this check?

File: cli/htmgo/tasks/astgen/entry.go [hunk 2]
```diff
@@ -390,9 +391,20 @@ func writePagesFile() {
...
+	_, err := os.Stat(path)
+	return !os.IsNotExist(err)
+}
+
 func GetModuleName() string {
 	wd := process.GetWorkingDir()
 	modPath := filepath.Join(wd, "go.mod")
+
+	if HasModuleFile(modPath) == false {
+		fmt.Fprintf(os.Stderr, "Module not found: go.mod file does not exists.")
```
Comment (maddalax) [inline cli/htmgo/tasks/astgen/entry.go:404]: "does not exist." instead of "does not exists"

File: cli/htmgo/tasks/astgen/entry.go (hunk 1 above)
Comment (rafaelDev0ps) [reply cli/htmgo/tasks/astgen/entry.go:423]: so the project directory check is not required anymore or should we use a different approach?

Comment (maddalax) [reply cli/htmgo/tasks/astgen/entry.go]: ah I see, sorry I didn't explain it at all in the GH issue. By that I meant make sure basically that there is:

1. go.mod
2. pages directory
3. partials directory

basically ensure its a valid project structure

Comment (rafaelDev0ps) [reply cli/htmgo/tasks/astgen/entry.go]: Ohhh gotcha, let me implement that then, thanks for clarifying

**Review by maddalax — APPROVED** [review]
lgtm 👍 , thank you

**Comment by rafaelDev0ps** [comment]
@maddalax added pages and partials directores checks, I believe this covers all of the issue's request.

**Review by maddalax — APPROVED** [review]

**Comment by rafaelDev0ps** [comment]
damn I thought you merged this PR, I messed the pull request, I'll close this one and open other one...

**Pull request closed without merging**